src/OmniSharp/Dnx/DnxSdk.cs

```diff
@@ -0,0 +1,184 @@
+
+using System;
+using System.IO;
+using System.Text;
+
+namespace OmniSharp.Dnx
+{
+    public class DnxSdk
+    {
+        public static string GetRuntimeNameFromFullPath(string runtimePath)
+        {
+            // Treat the folder name as the runtime name
+            // ~/.dnx/runtimes/{runtimename}/
+            var slash = runtimePath.TrimEnd(Path.DirectorySeparatorChar).LastIndexOf(Path.DirectorySeparatorChar);
+
+            return runtimePath.Substring(slash + 1);
+        }
+
+        public static string GetFullName(string version, string flavor, string os, string arch)
+        {
+            return GetRuntimeName(flavor, os, arch) + $".{version}";
+        }
+
+        public static bool TryParseFullName(string fullName, out string flavor, out string os, out string arch, out string version)
```

Comment:
Which results in

![image](https://cloud.githubusercontent.com/assets/1269157/10272616/67cd3e9a-6af0-11e5-844e-550d1355e6fb.png)